### PR TITLE
build: added build module

### LINKED ISSUE #40: Implement `tt build` module
The module will be used to build an application for local development.

### PR DESCRIPTION
**Pull request by psergee**
Added build module with local build implementation. --spec option is supported to specify rockspec file to use.

Closes #40

### CODE REVIEW

File: cli/util/execute.go [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+	"github.com/mattn/go-isatty"
+
+	"github.com/apex/log"
+	"github.com/briandowns/spinner"
+)
+
+type emptyStruct struct{}
+
+// readyChannel is a channel used to signal completion of command execution.
+type readyChannel chan emptyStruct
```
Comment (vr009) [inline cli/util/execute.go:23]: I would rename it to readyChan, sounds clear and more compactly and similar with chan type. But I don't insist, just idea. Feel free to ignore.

**Review by vr009 — APPROVED** [review]
LGTM. Some moments were discussed by voice.

File: cli/util/execute.go [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+
+// IsExecOwner checks if specified file has owner execute permissions.
+func IsExecOwner(path string) (bool, error) {
+	fileInfo, err := os.Stat(path)
+	if err != nil {
+		return false, err
+	}
+
+	perm := fileInfo.Mode().Perm()
+	return perm&execOwnerPerm != 0, nil
```
Comment (0x501D) [inline cli/util/execute.go:152]: Nit: Maybe we shoult add functions for bit manipulation to utils: like this:
```go
func BitSet(b, flag Bits) Bits    { return b | flag }
func BitClear(b, flag Bits) Bits  { return b &^ flag }
func BitToggle(b, flag Bits) Bits { return b ^ flag }
func BitHas(b, flag Bits) bool    { return b&flag != 0 }
```

Comment (psergee) [reply cli/util/execute.go]: I have added only bit set check function for 32-bit uints. Other functions are not required yet.

**Pull request merged**